### ISSUE TITLE
[v0.9] Could not parse expression variables device

### ISSUE THREAD

**Issue by sweebee** [pimatic/pimatic#830]
I'm getting this error with the variables device. The device uses variables of an other device.

```
21:34:50.774 [pimatic] error: Could not get attribute value of Vandaag.vandaag: Could not parse expression
21:34:51.016 [pimatic] debug: Error: Could not parse expression
```

device attr:
```
{
          "name": "vandaag",
          "expression": "$kwh-meter.kWh - $kw24",
          "type": "number",
          "unit": "kWh",
          "acronym": "Verbruik"
        },
```

**Comment by sweetpi** [comment]
Du you have may previous errors, so that the device didn't load?

**Comment by sweebee** [comment]
The kwh meter variables do update fine. Names of the variables match the variables of the kwh device.

Full error:
```
22:34:33.798 [pimatic] debug: Error: Could not parse expression
22:34:33.798 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:33.798 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:33.798 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:33.798 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:33.798 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:33.798 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:33.798 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:33.798 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:33.798 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:33.798 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:33.798 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:33.802 [pimatic] error: Could not get attribute value of Vandaag.Min: Could not parse expression
22:34:33.809 [pimatic] debug: Error: Could not parse expression
22:34:33.809 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:33.809 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:33.809 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:33.809 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:33.809 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:33.809 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:33.809 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:33.809 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:33.809 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:33.809 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:33.809 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:33.813 [pimatic] error: Could not get attribute value of Vandaag.Max: Could not parse expression
22:34:33.820 [pimatic] debug: Error: Could not parse expression
22:34:33.820 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:33.820 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:33.820 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:33.820 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:33.820 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:33.820 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:33.820 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:33.820 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:33.820 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:33.820 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:33.820 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:33.824 [pimatic] error: Could not get attribute value of Vandaag.winst: Could not parse expression
22:34:33.831 [pimatic] debug: Error: Could not parse expression
22:34:33.831 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:33.831 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:33.831 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:33.831 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:33.831 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:33.831 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:33.831 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:33.831 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:33.831 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:33.831 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:33.831 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:34.134 [pimatic] warn: Could not update variable kwh-vandaag.vandaag: Could not parse expression
22:34:34.142 [pimatic] debug: Error: Could not parse expression
22:34:34.142 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:34.142 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:34.142 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:34.142 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:34.142 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:34.142 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:34.142 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:34.142 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:34.142 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:34.142 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:34.142 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:34.146 [pimatic] warn: Could not update variable kwh-vandaag.Min: Could not parse expression
22:34:34.153 [pimatic] debug: Error: Could not parse expression
22:34:34.153 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:34.153 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:34.153 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:34.153 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:34.153 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:34.153 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:34.153 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:34.153 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:34.153 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:34.153 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:34.153 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:34.156 [pimatic] warn: Could not update variable kwh-vandaag.Max: Could not parse expression
22:34:34.163 [pimatic] debug: Error: Could not parse expression
22:34:34.163 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:34.163 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:34.163 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:34.163 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:34.163 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:34.163 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:34.163 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:34.163 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:34.163 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:34.163 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:34.163 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
22:34:34.167 [pimatic] warn: Could not update variable kwh-vandaag.winst: Could not parse expression
22:34:34.174 [pimatic] debug: Error: Could not parse expression
22:34:34.174 [pimatic] debug:>    at VariableManager.parseVariableExpression (/home/pi/pimatic-app/node_modules/pimatic/lib/variables.coffee:366:19)
22:34:34.174 [pimatic] debug:>    at parseExprAndAddListener (/home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:672:27)
22:34:34.174 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic/lib/devices.coffee:697:17
22:34:34.174 [pimatic] debug:>    at tryCatcher (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/util.js:16:23)
22:34:34.174 [pimatic] debug:>    at Promise._settlePromiseFromHandler (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:503:31)
22:34:34.174 [pimatic] debug:>    at Promise._settlePromise (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:560:18)
22:34:34.174 [pimatic] debug:>    at Promise._settlePromise0 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:605:10)
22:34:34.174 [pimatic] debug:>    at Promise._settlePromises (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:684:18)
22:34:34.174 [pimatic] debug:>    at Promise._fulfill (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/promise.js:629:18)
22:34:34.174 [pimatic] debug:>    at [object Object]._onTimeout (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/release/timers.js:26:46)
22:34:34.174 [pimatic] debug:>    at Timer.listOnTimeout (timers.js:89:15)
```

**Comment by sweebee** [comment]
After recreating the device I couldn't reproduce it anymore.

**Comment by sweebee** [comment]
To soon, after a reboot the  problem is back again. 

When i edit the device and click save it works again.

**Comment by sweetpi** [comment]
It was a race condition with variable initialization. Fixed in latest v0.9 beta.
